pkg/validate/dynamic/dynamic.go

```diff
@@ -25,28 +25,34 @@ import (
-	errMsgNSGNotAttached                    = "The provided subnet '%s' is invalid: must have a network security group attached."
-	errMsgNSGNotProperlyAttached            = "When the enable-preconfigured-nsg option is specified, both the master and worker subnets should have network security groups (NSG) attached to them before starting the cluster installation."
-	errMsgSPHasNoRequiredPermissionsOnNSG   = "The %s service principal (Application ID: %s) does not have Network Contributor role on network security group '%s'. This is required when the enable-preconfigured-nsg option is specified."
-	errMsgSubnetNotFound                    = "The provided subnet '%s' could not be found."
-	errMsgSubnetNotInSucceededState         = "The provided subnet '%s' is not in a Succeeded state"
-	errMsgSubnetInvalidSize                 = "The provided subnet '%s' is invalid: must be /27 or larger."
-	errMsgSPHasNoRequiredPermissionsOnVNet  = "The %s service principal (Application ID: %s) does not have Network Contributor role on vnet '%s'."
-	errMsgVnetNotFound                      = "The vnet '%s' could not be found."
-	errMsgSPHasNoRequiredPermissionsOnRT    = "The %s service principal does not have Network Contributor role on route table '%s'."
-	errMsgRTNotFound                        = "The route table '%s' could not be found."
-	errMsgSPHasNoRequiredPermissionsOnNatGW = "The %s service principal does not have Network Contributor role on nat gateway '%s'."
-	errMsgNatGWNotFound                     = "The nat gateway '%s' could not be found."
-	errMsgCIDROverlaps                      = "The provided CIDRs must not overlap: '%s'."
-	errMsgInvalidVNetLocation               = "The vnet location '%s' must match the cluster location '%s'."
+	errMsgSPHasMissingReaderRole               = "The %s service principal (Application ID: %s) does not have Reader permission"
+	errMsgSPHasMissingNetworkContributorRole   = "The %s service principal (Application ID: %s) does not have Network Contributor role"
+	errMsgPlatformIdentityHasMissingPermission = "The %s platform managed identity does not have required permissions"
+	errMsgNSGAttached                          = "The provided subnet '%s' is invalid: must not have a network security group attached."
+	errMsgOriginalNSGNotAttached               = "The provided subnet '%s' is invalid: must have network security group '%s' attached."
+	errMsgNSGNotAttached                       = "The provided subnet '%s' is invalid: must have a network security group attached."
+	errMsgNSGNotProperlyAttached               = "When the enable-preconfigured-nsg option is specified, both the master and worker subnets should have network security groups (NSG) attached to them before starting the cluster installation."
+	errMsgMissingPermissionsOnNSG              = "on network security group '%s'. This is required when the enable-preconfigured-nsg option is specified."
```

Comment:
This error message was truncated, it looks like it was a mistake? In my opinion we should just maintain separate error messages for "CSP missing required permissions" vs "managed identity missing required permissions", or just maintain a generic "the identity 'foo' is missing required permissions" for both cases.

Reply:
Working on this now.

Reply:
Separated error messaging based on the authorizerType.